corehq/sql_db/tests/utils.py

```diff
@@ -0,0 +1,31 @@
+from django.conf import settings
+from sqlalchemy import create_engine
+from sqlalchemy.exc import ProgrammingError
+
+from corehq.util.decorators import ContextDecorator
+
+
+class database_creator(ContextDecorator):
```

Comment:
this is neat. i wonder if it should be named something that also makes it clear it's being deleted? e.g. `temporary_new_database` or something